### DISCUSSION TITLE
cider-macroexpand-1 not working

### DISCUSSION THREAD

**Discussion by stevemolitor** [babashka/babashka#1613] [Ideas]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

**version**

babashka v1.3.182

**platform**

MacOS Ventura 13.5

**problem**

When I run `(cider-macroexpand-1)` / `"C-c RET"` in Emacs with cider on a form I get this error message:

```
cider-ensure-op-supported: ‘cider-macroexpand-1’ requires the nREPL op "macroexpand" (provided by 
```

Is this a known limitation? It works for me in vanilla clojure + cider, but not with bb.

**repro**

From within Emacs, jack into a bb cider session. Invoke `"C-c RET"` at the end of this expression:

```clojure
(-> (str "hi ") (str "there"))
```

**expected behavior**

Cider should show me the expanded macro.

**Comment by borkdude** [comment]
The babashka nREPL server doesn't implement all CIDER ops. CIDER is effectively nREPL middleware and not part of nREPL proper. 
I suggest just using `(macroexpand-1 ...)` in a REPL instead.
If you want to work with JVM tooling, a handy way to get from a bb project to a JVM project is:

```
$ bb print-deps > deps.edn
```

and then just cider-jack-in to your JVM project.

**Reply by stevemolitor** [reply]
Perfect thx!
